src/vue/common/OperationRecord.vue

```diff
@@ -0,0 +1,215 @@
+          <span class="operation-record__date">
+            {{ operation.date | formatCalendar }}
+          </span>
+        </td>
+        <td>
+          {{ `operation-names.${operation.typeI}` | globalize }}
+        </td>
+        <td>
+          <span
+            :class="{'operation-record__negative-amount'
+              : accountId === operation.sender}">
+            {{
+              { value: operation.amount, currency: operation.asset } |
+                formatMoney
+            }}
+          </span>
+        </td>
+        <td class="operation-record__counterparty">
+          {{ operation.counterparty }}
+        </td>
+        <td class="operation-record__status">
+          <span :class="`operation-record__status--${operation.state}`">
+            {{ operation.state }}
+          </span>
+        </td>
+        <td class="operation-record__td-btn">
+          <button
+            class="operation-record__btn"
+            @click="isOpenDetails = !isOpenDetails">
+            <i
+              v-if="isOpenDetails"
+              class="mdi mdi-chevron-up" />
+            <i
+              v-else
+              class="mdi mdi-chevron-down" />
+          </button>
+        </td>
+      </tr>
+    </table>
+    <template v-if="isOpenDetails">
+      <details-issuance
+        v-if="OP_TYPES.createIssuanceRequest === operation.typeI"
+        :operation="operation" />
+      <payment-details
+        v-if="OP_TYPES.paymentV2 === operation.typeI ||
+          OP_TYPES.payment === operation.typeI"
+        :operation="operation" />
+      <match-details
+        v-if="OP_TYPES.manageOffer === operation.typeI"
+        :operation="operation" />
+      <match-details
+        v-if="OP_TYPES.checkSaleState === operation.typeI"
+        :operation="operation" />
+      <withdrawal-details
+        v-if="OP_TYPES.createWithdrawalRequest === operation.typeI"
+        :operation="operation" />
+    </template>
+  </div>
+</template>
+
+<script>
+import DetailsIssuance from './OperationDetails/Issuance.Details'
+import PaymentDetails from './OperationDetails/Payment.Details'
+import MatchDetails from './OperationDetails/Match.Details'
+import WithdrawalDetails from './OperationDetails/Withdrawal.Details'
+import { mapGetters } from 'vuex'
+import { vuexTypes } from '@/vuex/types'
+import { RecordWrapper } from '@/js/records'
+import { OP_TYPES } from '@tokend/js-sdk'
+
+export default {
+  name: '',
+  components: {
+    DetailsIssuance,
+    PaymentDetails,
+    MatchDetails,
+    WithdrawalDetails
+  },
+  props: {
+    operationData: { type: Object, required: true }
+  },
+  data: _ => ({
+    OP_TYPES,
+    operation: {},
+    isOpenDetails: false
+  }),
+  computed: {
+    ...mapGetters([
+      vuexTypes.accountId
+    ])
+  },
+  watch: {
+    operationData: {
+      immediate: true,
+      handler: 'parsOperation'
+    }
+  },
+  methods: {
+    parsOperation () {
+      this.operation = RecordWrapper.operation(this.operationData, {
```

Comment:
Why don't wrap all the operation in the parent component, so you won't need to have a watcher here at all